pubsublite/internal/telemetry/consumer_test.go

```diff
@@ -0,0 +1,171 @@
+					SpanKind: trace.SpanKindConsumer,
+					Attributes: []attribute.KeyValue{
+						semconv.MessagingSystemKey.String("pubsublite"),
+						semconv.MessagingSourceKindTopic,
+						semconv.MessagingOperationProcess,
+						semconv.MessagingMessageIDKey.String(""),
+					},
+					InstrumentationLibrary: instrumentation.Library{
+						Name: "test",
+					},
+				},
+			},
+		},
+		{
+			name: "with attributes that don't match a parent span",
+			msg: &pubsub.Message{
+				Attributes: map[string]string{
+					"hello": "world",
+				},
+			},
+			expectedSpans: tracetest.SpanStubs{
+				tracetest.SpanStub{
+					Name:     "pubsublite.Receive",
+					SpanKind: trace.SpanKindConsumer,
+					Attributes: []attribute.KeyValue{
+						semconv.MessagingSystemKey.String("pubsublite"),
+						semconv.MessagingSourceKindTopic,
+						semconv.MessagingOperationProcess,
+						semconv.MessagingMessageIDKey.String(""),
+					},
+					InstrumentationLibrary: instrumentation.Library{
+						Name: "test",
+					},
+				},
+			},
+		},
+		{
+			name: "with attributes that matches a parent span",
+			msg: &pubsub.Message{
+				Attributes: map[string]string{
+					"traceparent": "00-4bf92f3577b34da6a3ce929d0e0e4736-00f067aa0ba902b7-01",
+				},
+			},
+			traceID: mustTraceIDFromHex("4bf92f3577b34da6a3ce929d0e0e4736"),
+			expectedSpans: tracetest.SpanStubs{
+				tracetest.SpanStub{
+					Name:     "pubsublite.Receive",
+					SpanKind: trace.SpanKindConsumer,
+					SpanContext: trace.NewSpanContext(trace.SpanContextConfig{
+						TraceID: mustTraceIDFromHex("4bf92f3577b34da6a3ce929d0e0e4736"),
+					}),
+					Parent: trace.NewSpanContext(trace.SpanContextConfig{
+						TraceID: mustTraceIDFromHex("4bf92f3577b34da6a3ce929d0e0e4736"),
+						Remote:  true,
+					}),
+					Attributes: []attribute.KeyValue{
+						semconv.MessagingSystemKey.String("pubsublite"),
+						semconv.MessagingSourceKindTopic,
+						semconv.MessagingOperationProcess,
+						semconv.MessagingMessageIDKey.String(""),
+					},
+					InstrumentationLibrary: instrumentation.Library{
+						Name: "test",
+					},
+				},
+			},
+		},
+	} {
+		t.Run(tt.name, func(t *testing.T) {
+			h := Consumer(tp.Tracer("test"), func(ctx context.Context, msg *pubsub.Message) {
+				// No need to do anything here
+			})
+
+			h(context.Background(), tt.msg)
+
+			spans := exp.GetSpans()
+			for i := range spans {
+				// Nullify data we don't use/can't set manually
+				spans[i].SpanContext = spans[i].SpanContext.
+					WithTraceID(tt.traceID).
+					WithSpanID(tt.spanID).
+					WithTraceFlags(0)
+				spans[i].Parent = spans[i].Parent.
+					WithTraceID(tt.traceID).
+					WithSpanID(tt.parentSpanID).
+					WithTraceFlags(0)
+				spans[i].Resource = nil
+				spans[i].StartTime = time.Time{}
+				spans[i].EndTime = time.Time{}
+			}
+
+			assert.Equal(t, spans, tt.expectedSpans)
+
+			// Reset for next tests
+			exp.Reset()
+		})
+	}
+}
+
+func mustTraceIDFromHex(s string) (t trace.TraceID) {
```

Comment:
issue: we need a `t.Helper` here to print the correct line number in case of failure